src/pages/index.js

```diff
@@ -1,46 +1,62 @@
 //Import styles
 import "./index.css";
 
-//Import classes
+//Import classes, settings and API
 import { FormValidator } from "../components/FormValidator.js";
 import { Card } from "../components/Card.js";
 import { Section } from "../components/Section.js";
 import { UserInfo } from "../components/UserInfo.js";
 import { PopupWithImage } from "../components/PopupWithImage.js";
 import { PopupWithForm } from "../components/PopupWithForm.js";
+import { PopupDelete } from "../components/PopupDelete.js";
+import { validationSettings, userSettings } from "../utils/constants.js";
+import { Api } from "../utils/Api.js";
 
-//Import settings
-import {
-  initialCards,
-  validationSettings,
-  userSettings,
-} from "../utils/constants.js";
-
-//Create UserInfo class
-const userInfo = new UserInfo(userSettings);
+//Create API
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "a50cd7e6-b371-411e-bfcd-861efcd7c545",
+    "Content-Type": "application/json",
+  },
+});
 
-//Selecting buttons
+//Selecting elements
 const buttonEdit = document.querySelector(".profile__button_type_edit");
 const buttonAdd = document.querySelector(".profile__button_type_add");
+const avatar = document.querySelector(".profile__image");
+const avatarEdit = document.querySelector(".image__container");
+
+//Create sectionCard class
+const sectionCard = new Section({ renderer: createCard }, ".cards__list");
+
+//Create UserInfo class
+const userInfo = new UserInfo(userSettings);
 
 //Create popups classes and set event listeners
 const popupImage = new PopupWithImage("#image-modal-id");
 const popupEdit = new PopupWithForm("#edit-modal-id", handleEditSumbit);
 const popupAdd = new PopupWithForm("#add-modal-id", handleAddSumbit);
-
-//Create card section class
-const sectionCard = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
-);
+const popupAvatar = new PopupWithForm("#avatar-modal-id", handleAvatarSubmit);
+const popupDelete = new PopupDelete("#delete-modal-id", handleDeleteSubmit);
 
 //Create validators
 const formEditValidator = new FormValidator(validationSettings, popupEdit.form);
 const formAddValidator = new FormValidator(validationSettings, popupAdd.form);
+const formAvatarValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```